File: app/src/main/java/com/sjn/stamp/db/dao/BaseDao.java
package com.sjn.stamp.db.dao;

import io.realm.Realm;

public abstract class BaseDao {

    protected Long getAutoIncrementId(Realm realm, Class clazz) {
        Number maxId = realm.where(clazz).max("id");
        if (maxId != null) {
            return maxId.intValue() + 1L;
        }
        return 1L;
    }
}

File: app/src/main/java/com/sjn/stamp/model/dao/BaseDao.kt
package com.sjn.stamp.model.dao

import io.realm.Realm
import io.realm.RealmModel
import java.lang.reflect.ParameterizedType

abstract class BaseDao<T : RealmModel> {

    fun getAutoIncrementId(realm: Realm): Long {
        val entityClass = (this.javaClass.genericSuperclass as ParameterizedType).actualTypeArguments[0]
        val maxId = realm.where(entityClass as Class<T>).max("id")
        return if (maxId != null) {
            maxId.toInt() + 1L
        } else 1L
    }
}
